Force emulating readonly disks as SCSI

IDE and AHCI emulation in qemu-xen does not support read-only disks.

-----------
diff --git a/tools/libs/light/libxl_dm.c b/tools/libs/light/libxl_dm.c
index 49d46e06..75340ff1 100644
--- a/tools/libs/light/libxl_dm.c
+++ b/tools/libs/light/libxl_dm.c
@@ -1363,7 +1363,7 @@ static int libxl__build_device_model_args_new(libxl__gc *gc,
             if (disks[i].is_cdrom) {
                 continue;
             }
-            if (strncmp(disks[i].vdev, "sd", 2) == 0) {
+            if (strncmp(disks[i].vdev, "sd", 2) == 0 || !disks[i].readwrite) {
                 flexarray_vappend(dm_args, "-device", "lsi53c895a", NULL);
                 break;
             }
@@ -1936,7 +1936,7 @@ static int libxl__build_device_model_args_new(libxl__gc *gc,
                     colo_mode = LIBXL__COLO_NONE;
                 }
 
-                if (strncmp(disks[i].vdev, "sd", 2) == 0) {
+                if (strncmp(disks[i].vdev, "sd", 2) == 0 || !disks[i].readwrite) {
                     const char *drive_id;
                     if (colo_mode == LIBXL__COLO_SECONDARY) {
                         drive = libxl__sprintf
-----------------
